typed-racket-lib/typed-racket/typecheck/tc-structs.rkt

```diff
@@ -272,6 +283,49 @@
 
   (parsed-struct sty names desc (struct-info-property nm/par) type-only))
 
+;; check and register types for a prefab struct
+(define (tc/prefab vars nm/par fld-names tys
```

Comment:
I don't think we need `#:type-only?` here because it's only used for `require/typed`. Since prefab structs can be declared anywhere, there's no need to do a special `require/typed` for them.